`tk init` panics during SSH validation

/kind bug

**What happened**:
When running `tarmak init` with no `~/.tarmak`, Tarmak panics.

**What you expected to happen**:
`tk init` to output a completed yaml in `~/.tarmak`

**How to reproduce it (as minimally and precisely as possible)**:
Stack trace below:
```
In which region should this environment reside?
  1) ap-northeast-1
  2) ap-northeast-2
  3) ap-south-1
  4) ap-southeast-1
  5) ap-southeast-2
  6) ca-central-1
  7) eu-central-1
  8) eu-west-1
  9) eu-west-2
 10) eu-west-3
 11) sa-east-1
 12) us-east-1
 13) us-east-2
 14) us-west-1
 15) us-west-2
> 8

name: env3fri
location: eu-west-1
provider: amazon[prov3fri]

Continue with this environment? [Y/n] 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x173fef2]

goroutine 1 [running]:
github.com/jetstack/tarmak/pkg/tarmak/ssh.(*SSH).Validate(0xc4209473b0, 0x453fe60, 0xc4209473b0)
	/home/alljames/go/src/github.com/jetstack/tarmak/pkg/tarmak/ssh/ssh.go:127 +0x52
github.com/jetstack/tarmak/pkg/tarmak/environment.(*Environment).Validate(0xc42047ea00, 0xc420ecd6f0, 0x1)
	/home/alljames/go/src/github.com/jetstack/tarmak/pkg/tarmak/environment/environment.go:270 +0xae
github.com/jetstack/tarmak/pkg/tarmak/initialize.(*Initialize).InitEnvironment(0xc420472540, 0xc4205918c0, 0x1, 0x1, 0x0)
	/home/alljames/go/src/github.com/jetstack/tarmak/pkg/tarmak/initialize/initialize.go:186 +0x619
github.com/jetstack/tarmak/pkg/tarmak/initialize.(*Initialize).GetEnvironment(0xc420472540, 0x0, 0xc420591978, 0x40fb56, 0x7fc8c46814c8)
	/home/alljames/go/src/github.com/jetstack/tarmak/pkg/tarmak/initialize/initialize.go:388 +0x6d2
github.com/jetstack/tarmak/pkg/tarmak/initialize.(*Initialize).InitCluster(0xc420472540, 0xc42056e480, 0x451d3e0, 0xc4200ce000, 0x450f160)
	/home/alljames/go/src/github.com/jetstack/tarmak/pkg/tarmak/initialize/initialize.go:272 +0x76
github.com/jetstack/tarmak/pkg/tarmak.(*Tarmak).CmdClusterInit(0xc42056e480, 0xc42056e480, 0x0)
	/home/alljames/go/src/github.com/jetstack/tarmak/pkg/tarmak/tarmak.go:187 +0xb0
github.com/jetstack/tarmak/cmd/tarmak/cmd.glob..func11(0x7760640, 0x77b7920, 0x0, 0x0)
	/home/alljames/go/src/github.com/jetstack/tarmak/cmd/tarmak/cmd/cluster_init.go:17 +0x4b
github.com/jetstack/tarmak/vendor/github.com/spf13/cobra.(*Command).execute(0x7760640, 0x77b7920, 0x0, 0x0, 0x7760640, 0x77b7920)
	/home/alljames/go/src/github.com/jetstack/tarmak/vendor/github.com/spf13/cobra/command.go:766 +0x2c1
github.com/jetstack/tarmak/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x7762ea0, 0x7ffe74746e2c, 0xc4200b8030, 0xc420591dd8)
	/home/alljames/go/src/github.com/jetstack/tarmak/vendor/github.com/spf13/cobra/command.go:852 +0x30a
github.com/jetstack/tarmak/vendor/github.com/spf13/cobra.(*Command).Execute(0x7762ea0, 0xc4200b8030, 0x1)
	/home/alljames/go/src/github.com/jetstack/tarmak/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/jetstack/tarmak/cmd/tarmak/cmd.Execute(0xc4200b8030, 0x1, 0x1)
	/home/alljames/go/src/github.com/jetstack/tarmak/cmd/tarmak/cmd/root.go:58 +0x166
main.main()
	/home/alljames/go/src/github.com/jetstack/tarmak/cmd/tarmak/main.go:22 +0x10b
```

**Anything else we need to know?**:
Could likely be remedied by a check for `nil`

**Environment**:
- Kubernetes version (use `kubectl version`): 
- Cloud provider or hardware configuration**: AWS
- Install tools:
- Others:

--- a/pkg/tarmak/ssh/ssh.go
+++ b/pkg/tarmak/ssh/ssh.go
@@ -124,6 +124,11 @@ func (s *SSH) Execute(host string, command string, argsAdditional []string) (ret
 }
 
 func (s *SSH) Validate() error {
+	// no environment in tarmak so we have no SSH to validate
+	if s.tarmak.Environment() == nil {
+		return nil
+	}
+
 	keyPath := s.tarmak.Environment().SSHPrivateKeyPath()
 	f, err := os.Stat(keyPath)
 	if err != nil {

--- a/pkg/tarmak/tarmak.go
+++ b/pkg/tarmak/tarmak.go
@@ -207,6 +207,11 @@ func (t *Tarmak) CmdClusterInit() error {
 	}
 	t.log.Infof("successfully initialized cluster '%s'", cluster.ClusterName())
 
+	err = t.Config().SetCurrentCluster(cluster.ClusterName())
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 

